### ISSUE TITLE
Explore how to limit the amount of time we spend compiling a Hedy program

### ISSUE THREAD

**Issue by fpereiro** [Felienne/hedy#461]
Currently, a large program (or a hard to parse program) can block all processing within the same thread for all other users. It'd be great to explore how we can limit this to a fixed amount of time. This can improve performance and mitigate potential DDoS through execution of programs.

**Comment by rix0rrr** [comment]
While this is true, I wouldn't invest too much time into this yet. I feel we don't have definitive proof yet to show that the parsing is really the reason the server goes down sometime.

Let's wait for the details of the new logging to come in to see if we can tease out the actual bottlenecks a bit better.

**Comment by fpereiro** [comment]
@rix0rrr understood, I'll close this now and wait until we have a clearer view of performance bottlenecks.
